### PR TITLE
[FLINK-13043][Library / CEP] Fix the bug of parsing Dewey number from string

### PR DESCRIPTION
**Pull request by liyafan82**
## What is the purpose of the change

There is a bug in the current implementation for parsing the Dewey number:

 

String[] splits = deweyNumberString.split("\\.");

if (splits.length == 0){
 return new DeweyNumber(Integer.parseInt(deweyNumberString)); 
}
 

The length in the if condition should be 1 instead of 0.

## Brief change log

  - Fix the problem by modifying the implementation of DeweyNumber#fromString
  

## Verifying this change

This change is already covered by existing tests, such as DeweyNumberTest.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/DeweyNumber.java [hunk 1]
```diff
@@ -175,16 +175,18 @@ public String toString() {
 	public static DeweyNumber fromString(final String deweyNumberString) {
 		String[] splits = deweyNumberString.split("\\.");
 
-		if (splits.length == 0) {
+		if (splits.length == 1) {
 			return new DeweyNumber(Integer.parseInt(deweyNumberString));
-		} else {
+		} else if (splits.length > 0) {
```
Comment (Myasuka) [inline flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/DeweyNumber.java]: In which case would the returned `splits` has a length not larger than `0`? I think the condition `if (splits.length > 0)` might be redundant.

Comment (liyafan82) [reply flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/DeweyNumber.java]: @Myasuka thanks for the comments. 
In some rare cases, the length can be 0. So this condition guards against such cases.

**Comment by StephanEwen** [comment]
This fix needs a test before it can be merged...

**Comment by StephanEwen** [comment]
@flinkbot approve-until architecture

**Comment by liyafan82** [comment]
> This fix needs a test before it can be merged...

Test added. Please take a look. Thanks.

**Comment by StephanEwen** [comment]
@flinkbot approve all

**Comment by StephanEwen** [comment]
Looks good, merging this...

**Pull request closed without merging**